Warnings in sockets provider

FWIW, I see these warnings in the sockets provider on OS X:

```
  CC       prov/sockets/src/src_libfabric_la-sock_fabric.lo
prov/sockets/src/sock_fabric.c:259:22: warning: incompatible pointer types
      passing 'struct sockaddr_in *' to parameter of type
      'const struct sockaddr *' [-Wincompatible-pointer-types]
        ret = connect(sock, dest_addr, len);
                            ^~~~~~~~~
/usr/include/sys/socket.h:563:41: note: passing argument to parameter here
int     connect(int, const struct sockaddr *, socklen_t) __DARWIN_ALIAS_...
                                            ^
1 warning generated.
  CC       prov/sockets/src/src_libfabric_la-sock_conn.lo
prov/sockets/src/sock_conn.c:210:7: warning: variable 'optval' is uninitialized
      when used here [-Wuninitialized]
                                                optval, strerror(optval));
                                                ^~~~~~
prov/sockets/src/sock_util.h:52:53: note: expanded from macro 'SOCK_LOG_ERROR'
#define SOCK_LOG_ERROR(...) FI_WARN(sock_prov_name, __VA_ARGS__)
                                                    ^
./include/fi_log.h:52:47: note: expanded from macro 'FI_WARN'
#define FI_WARN(prov, ...) fi_warn_impl(prov, __VA_ARGS__)
                                              ^
prov/sockets/src/sock_conn.c:174:21: note: initialize the variable 'optval' to
      silence this warning
        int conn_fd, optval, ret;
                           ^
                            = 0
1 warning generated.

--- a/prov/sockets/src/sock_comm.c
+++ b/prov/sockets/src/sock_comm.c
@@ -125,7 +125,7 @@ ssize_t sock_comm_send(struct sock_conn *conn, const void *buf, size_t len)
 
 ssize_t sock_comm_recv_socket(struct sock_conn *conn, void *buf, size_t len)
 {
-	int ret;
+	ssize_t ret;
 	
 	ret = read(conn->sock_fd, buf, len);
 	if (ret < 0) {

--- a/prov/sockets/src/sock_conn.c
+++ b/prov/sockets/src/sock_conn.c
@@ -171,7 +171,7 @@ uint16_t sock_conn_map_connect(struct sock_ep *ep,
 			       struct sock_conn_map *map, 
 			       struct sockaddr_in *addr)
 {
-	int conn_fd, optval, ret;
+	int conn_fd, optval = 0, ret;
 	char use_conn;
 	struct timeval tv;
 	socklen_t optlen;
@@ -407,7 +407,7 @@ int sock_conn_listen(struct sock_ep *ep)
 	if (atoi(listener->service) == 0) {
 		addr_size = sizeof(addr);
 		if (getsockname(listen_fd, (struct sockaddr *) &addr, &addr_size))
-			return -FI_EINVAL;
+			goto err;
 		snprintf(listener->service, sizeof listener->service, "%d",
 			 ntohs(addr.sin_port));
 		SOCK_LOG_INFO("Bound to port: %s\n", listener->service);

--- a/prov/sockets/src/sock_fabric.c
+++ b/prov/sockets/src/sock_fabric.c
@@ -256,7 +256,7 @@ static int sock_get_src_addr(struct sockaddr_in *dest_addr,
 		return -errno;
 
 	len = sizeof(*dest_addr);
-	ret = connect(sock, dest_addr, len);
+	ret = connect(sock, (struct sockaddr*)dest_addr, len);
 	if (ret) {
 		SOCK_LOG_ERROR("Failed to connect udp socket\n");
 		ret = -errno;

